src/heitech.zer0mqXt.core/utils/PollerUtils.cs

```diff
@@ -10,10 +11,9 @@ public static void DisposeOf(this NetMQPoller poller, NetMQSocket anySocket, Eve
             anySocket.ReceiveReady -= receiveDelegate;
             if (poller != null && poller.IsRunning)
             {
-                poller.Stop();
-                poller.Remove(anySocket);
-                poller.Dispose();
-                anySocket.Dispose();
+                // needs to be done from the Poller´s thread or else a race condition is encountered.
+                new Task(() => { poller.Remove(anySocket); anySocket.Dispose(); })
+                .Start(poller);
```

Comment:
Cool - I didn't know about the `TaskScheduler` overload of `Task.Start`.

Reply:
Yeah that one is quite useful for things with thread affinity like the Poller for example.